### ISSUE TITLE
Overfitting issue for detection task with small dataset

### ISSUE THREAD

**Issue by deJQK** [ultralytics/ultralytics#19748]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I am trying to use Yolov8 to train a detection task on [bone fracture detection from kaggle](https://www.kaggle.com/datasets/pkdarabi/bone-fracture-detection-computer-vision-project). However, I always encounters the overfitting issue, where the training loss continues to drop, yet the validation loss and final metrics increases after several epochs. I have tried to use smaller model size (e.g., Yolov8n), and also adjust the learning rate, but none of them seem to work. I wonder if there is any suggestion on this. Also, where could I check the learning rate used during training, say in any log file or anywhere? Thanks!

![Image](https://github.com/user-attachments/assets/53a5cb7c-57b8-4a58-9847-c5857bea5f5a)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @deJQK, thank you for sharing your question and for your interest in Ultralytics 🚀! We’re here to help. An Ultralytics engineer will review your issue soon, but in the meantime, here are some suggestions that might be helpful:

### General Guidance
We recommend reviewing our [Docs](https://docs.ultralytics.com/) for comprehensive tutorials, including [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/), which could help mitigate overfitting in small datasets. Adjustments such as data augmentation, dropout, or early stopping may also be worth exploring.

To check the learning rate used during training, you can find this information in the training logs or the `results*.txt` file generated during training. If you're not seeing this, ensure you are using the latest version of the `ultralytics` package by upgrading with:

```bash
pip install -U ultralytics
```

### Minimum Reproducible Example (MRE)
If this is related to a potential 🐛 Bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can better understand and debug the issue.

### Community Support
Join the Ultralytics community for additional support and discussions:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Reddit](https://reddit.com/r/Ultralytics) for community-driven insights

### Verified Environments
Ensure you are running YOLO in one of the following verified environments:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon AWS**: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Status
Check our CI status to ensure all tests are passing:  
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

We appreciate your patience and are confident that with these resources, you'll be able to make progress. If you have additional details, feel free to share them here. Stay tuned for further guidance from our team! 😊

**Comment by Petros626** [comment]
try data augmentation or change the weights of box and dfl.

**Comment by deJQK** [comment]
> try data augmentation or change the weights of box and dfl.

Hi @Petros626 , thanks a lot for your kind help. I tried to use augment after your reply, and also tried to increase or decrease the loss weights for box and dfl, but the behavior is still the same and the final mAP is similar. I also tried to increase weight decay (from the default 5e-4 to 5e-3), or use dropout (with 0.1 against the default value 0.0), but none of them seems to work. Do you have any idea on this? There are totally 3631 training and 348 validation data, respectively.

**Comment by Petros626** [comment]
Decrease the box and dfl loss slightly (0.5&.01 steps) when you're using data augmentation think about the sense of all possibles techniques. It depends on your dataset,which techniques make sense. Try multi scale, rect, mosaic.

**Comment by glenn-jocher** [comment]
To address overfitting with small datasets in YOLO11 detection tasks, consider these steps:  
1. **Advanced Augmentation**: Enable `mosaic=0.5` and `mixup=0.3` in your training config to increase data diversity while avoiding excessive distortion of medical images.  
2. **Multi-Scale Training**: Add `multi_scale=[0.5, 1.0]` to help the model generalize across scales.  
3. **Layer Freezing**: Start with `freeze=10` to freeze backbone layers initially, then unfreeze after 50 epochs.  
4. **Learning Rate Scheduling**: Use `cos_lr=True` for smoother transitions between learning rates.  

The learning rate trajectory can be visualized in TensorBoard logs (located in `runs/detect/train`) or parsed from the `results.csv` file. For medical imaging specifics, review our Medical Image Analysis Guide which covers specialized techniques for small datasets.  

For implementation details, refer to:  
- Training Configuration Options  
- Data Augmentation Parameters  
- Layer Freezing Guide  

Let us know which combination yields the best results for your fracture detection task! 🩻

**Comment by deJQK** [comment]
Thanks a lot @Petros626 and @glenn-jocher for your kind suggestions. I will try what you mentioned later. I just realized that the labels of my data are polygon bbox coordinates, not rectangular, and I was using yolov8n/s/m. I tried with yolov8n-obb, but the behavior is similar. I might need first preprocess my data to make it as rectangular bbox. Please kindly let me know if you think it is not necessary. (By polygon bbox, I mean the labels are like class_label x1 y1 x2 y2 ..., and by rectangular bbox, I mean labeling by class_label x_center y_center width height).

**Comment by deJQK** [comment]
Hi @glenn-jocher , I wonder how to set the multi_scale to a list? It seems that in the default yaml it is a bool variable?

**Comment by Petros626** [comment]
`multi_scale=True` . Here is my configuration:
```python
results = model.train(data='/home/heizung1/ultralytics_yolov8-obb_ob_kitti/ultralytics/cfg/datasets/kitti_bev.yaml', epochs=50,
                      time=None, patience=0, batch=4, imgsz=640, save=True, save_period=25, cache=True,
                      device=[0, 1], workers=8, project='kitti_bev_yolo', name='avoid_overfitting', exist_ok=False, 
                      pretrained=False, optimizer='Adam', seed=0, deterministic=False, single_cls=False,
                      classes=None, rect=False, multi_scale=True, cos_lr=False, close_mosaic=0, resume=False, amp=False,
                      fraction=1.0, profile=False, freeze=None, lr0=0.01, lrf=0.01,  momentum=0.937,
                      weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, 
                      box=7.0, cls=0.3, dfl=0.9, pose=0.0, kobj=0.0, nbs=64, overlap_mask=False, mask_ratio=0, # dflt: 7.5, 0.4, 1.0
                      dropout=0.0, val=True, plots=True,
                      
                      hsv_h=0.0, hsv_s=0.0, hsv_v=0.0, degrees=0.0, translate=0.0, scale=0.0, shear=0.0, perspective=0.0,
                      flipud=0.0, fliplr=0.0, bgr=0.0, mosaic=0.0, mixup=0.0, copy_paste=0.0, copy_paste_mode='mixup', auto_augment=None,
                      erasing=0.0, crop_fraction=0.0)
```

**Comment by glenn-jocher** [comment]
For `multi_scale`, it's a boolean flag (True/False) that toggles random image scaling +/-50% during training. To specify exact scale ranges, use `imgsz=[min, max]` instead. For OBB-specific guidance with polygon labels, see the Oriented Bounding Boxes documentation. 

Configuration details are in the Train Settings reference.

**Comment by deJQK** [comment]
Thank @Petros626 . I tried your config, but the results is quite bad. The mAP50 is less that 0.05 (shown below), although there indeed is no overfitting phenomenon. I thought this was due to not using pretrained model, but when I tried to use pretrained one, the results show similar behavior.

<img width="762" alt="Image" src="https://github.com/user-attachments/assets/b57ba214-be24-4bc1-b9eb-3edf515dca07" />

**Comment by Petros626** [comment]
As I said before, think about the sense of different techniques, copying my technique doesn't suit for you. You have to augment the data.

**Comment by glenn-jocher** [comment]
For OBB tasks with polygon labels, ensure your dataset follows the correct YOLO OBB format outlined in the Oriented Bounding Boxes documentation. Since you're seeing low mAP despite proper regularization, consider these focused steps:

1. Verify label conversions using the DOTA to YOLO-OBB conversion script
2. Gradually reintroduce augmentations like `mosaic=0.3` and `mixup=0.1` while monitoring stability
3. Try SGD optimizer with default `lr0=0.01` instead of Adam

The learning rate schedule can be visualized in your `results.csv` file or TensorBoard logs. Let us know which adjustments impact your fracture detection performance most significantly.
